Service/Tracing.php

```diff
@@ -8,6 +8,11 @@
 
 interface Tracing
 {
+    /**
+     * Injects necessary tracing headers into a an array.
+     */
+    public function injectTracingHeadersToArray(array $headers): void;
```

Comment:
I changed it to :
```
    /**
     * Injects necessary tracing headers into an array.
     * @param mixed $carrier
     * @return void
     *
     * @throws UnsupportedFormat when the format is not recognized by the tracer
     */
    public function injectTracingHeadersIntoCarrier(&$carrier): void;
```

As it is how the opentracing-php exposes it under the hood.

Reply:
That was the one thing I would argue against. Implicitely changing a passed argument within a method is widely considered a bad design. Explicitely returning the changed object would be the better way, and in this point I completely disagree with the underlying library implementation, which we should not try do model in our interfaces, but provide an opinionated API.
What do you think?

Reply:
I have no strong opinion here, but i would say that passing by value instead of passing by reference would more consistent in regard to the rest of the lib.
I'll change change it right away.

Reply:
done